test/Jaeger.Core.Tests/Reporters/RemoteReporterTests.cs

```diff
@@ -67,27 +67,15 @@ public void RemoteReporter_ShouldCallMetrics()
                 transport.Received(2).AppendAsync(span, Arg.Any<CancellationToken>());
             }
             transport.Received(1).CloseAsync(Arg.Any<CancellationToken>());
-            Assert.Equal(2, metrics.ReporterSuccess.Count);
-        }
-
-        [Fact]
-        public void RemoteReporter_ShouldCallMetricsFactory()
-        {
-            var transport = Substitute.For<ITransport>();
-            var metricsFactory = InMemoryMetricsFactory.Instance;
-
-            using (var reporter = new RemoteReporter.Builder(transport).WithMetricsFactory(metricsFactory).Build())
-            {
-                Assert.IsType<InMemoryMetricsFactory.InMemoryElement>(reporter._metrics.ReporterSuccess);
-            }
+            metrics.ReporterSuccess.Received(1).Inc(2);
```

Comment:
why not use in-memory implementation and just assert the final counter value? With a mock you are testing not only external side effect, i.e. the resulting value of the counter which is what we're interested in, but also _how it gets there_, which is mostly an irrelevant implementation detail that makes the test itself more brittle.